source/SH3/arc/types.cpp

```diff
@@ -151,7 +151,10 @@ int sh3_arc::LoadFile(const std::string& filename, std::vector<std::uint8_t>& bu
 
     static_cast<void>(index); // avoid clang warning - index *is* initialized now
 
-    std::ifstream sectionFile(section->sectionName, std::ios::binary);
+    std::string sectionPath = "data/" + section->sectionName;
+    sectionPath.insert(sectionPath.size() - 1, ".arc\0");
```

Comment:
The problem is there's a NUL terminator at the end of section->sectionName (part of the arc spec), which means the ".arc" part gets ignored